Allow aliasing with `:as` to rename imports

tvanhens opened the pull request:
There are classes with conflicting names in CDK. For example
`@aws-cdk/core.CfnParameter` and `@aws-cdk/aws-ssm.CfnParameter`. This
causes a conflict that is currently unresolvable.

This change adds the ability to change the name of an import with an
`:as` flag.

Example:

``` clojure
(import ["@aws-cdk/aws-ssm"
         CfnParameter :as CfnSsmParameter])
```

This would alias the `@aws-cdk/aws-ssm.CfnParameter` namespace and
refer the `@aws-cdk/aws-ssm.CfnParameter` class to the symbol
`CfnSsmParameter` in the current namespace.

Commit:
Allow aliasing with `:as` to rename imports

There are classes with conflicting names in CDK. For example
`@aws-cdk/core.CfnParameter` and `@aws-cdk/aws-ssm.CfnParameter`. This
causes a conflict that is currently unresolvable.

This change adds the ability to change the name of an import with an
`:as` flag.

Example:

``` clojure
(import ["@aws-cdk/aws-ssm"
         CfnParameter :as CfnSsmParameter])
```

This would alias the `@aws-cdk/aws-ssm.CfnParameter` namespace and
refer the `@aws-cdk/aws-ssm.CfnParameter` class to the symbol
`CfnSsmParameter` in the current namespace.

sinistral reviewed (approved):
Approved as the best option for a non-breaking change, but I think we should consider how we really want this to look.  This form is similar enough in both intent and structure that the differences to other more familiar forms of host interop might be confusing.

Commit:
Update syntax to better mirror node

Since `cdk/import` is serving as a way to import node modules, the
syntax of `cdk/import` has been changed to better reflect that.

For example, if you wanted to require the `App` class from
`@aws-cdk/core` in node you would write:

``` node
import { App } from "@aws-cdk/core"
```

The equivalent statement in cdk is now:

``` clojure
(cdk/import [[App] :from "@aws-cdk/core"])
```

The original syntax still works but will log a warning so this change
maintains backwards compatibility.

This syntax change was made for two reasons:

1. To better match the node syntax to make it easier to map to the CDK
   documentation.
2. To clearly demarcate the syntax from `clojure.core/import` and
   `clojure.core/require` as the behavior does not map to either.
